packages/flutter/lib/src/widgets/app.dart

```diff
@@ -149,6 +185,45 @@ class WidgetsApp extends StatefulWidget {
   /// for this application's [Localizations] widget.
   final Iterable<LocalizationsDelegate<dynamic>> localizationsDelegates;
 
+  /// This callback is responsible for choosing the app's locale
+  /// when the app is started, and when the user changes the
+  /// device's locale.
+  ///
+  /// The returned value becomes the locale of this app's [Localizations]
+  /// widget. The `oldLocale` parameter is null when the app starts. The
+  /// `newLocale` is the device's locale when the app started, or the
+  /// device locale the user selected after the app was started.
+  /// The `supportedLocales` parameter is just the value [supportedLocales].
+  ///
+  /// This callback must not be null. Its default value is
+  /// [defaultLocaleChangedHandler].
```

Comment:
You could also make defaultLocaleChangedHandler entirely private by saying that if the callback is null or returns null, you default to that behaviour. That removes a bunch of potential error cases.

Reply:
Good idea, less is more!